tests/lib/rules/no-duplicate-class-names.js

```diff
@@ -0,0 +1,233 @@
+/**
+ * @author Yizack Rangel
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+const RuleTester = require('../../eslint-compat').RuleTester
+const rule = require('../../../lib/rules/no-duplicate-class-names')
+
+const tester = new RuleTester({
+  languageOptions: {
+    parser: require('vue-eslint-parser'),
+    ecmaVersion: 2020,
+    sourceType: 'module'
+  }
+})
+
+tester.run('no-duplicate-class-names', rule, {
+  valid: [
+    {
+      filename: 'no-classes.vue',
+      code: '<template><div></div></template>'
+    },
+    {
+      filename: 'one-class.vue',
+      code: '<template><div class="foo"></div></template>'
+    },
+    {
+      filename: 'multiple-different-classes.vue',
+      code: '<template><div class="foo bar baz"></div></template>'
+    },
+    {
+      filename: 'no-duplicate-class-in-directive-string.vue',
+      code: `<template><div :class="'foo bar baz'"></div></template>`
+    },
+    {
+      filename: 'no-duplicate-class-in-directive-literal.vue',
+      code: '<template><div :class="`foo bar baz`"></div></template>'
+    },
+    {
+      filename: 'no-duplicate-class-in-directive-object.vue',
+      code: `<template><div :class="{ 'foo bar baz': true }"></div></template>`
+    },
+    {
+      filename: 'duplicate-class-and-directive.vue',
+      code: `<template><div class="foo" :class="'foo'"></div></template>`
```

Comment:
See https://github.com/vuejs/eslint-plugin-vue/issues/2932#issuecomment-3306121579 → let's discuss this there first.